test/nexus-integration.py

```diff
@@ -134,6 +135,63 @@ def test_search_nexus3_pyversion(self):
         )
         self.assertListEqual(versions, [pipio.Version('1.0.0'), pipio.Version('1.0.1rc1'), pipio.Version('1.0.1')])        
 
+class TestPut(unittest.TestCase):
+
+    def test_upload_package_for_pep440_compliant_version(self):
+        pkg_name = 'test_package2' # must differ from package names used in TestCheck
+        src_repo = os.path.join(THISDIR, 'generalized_package')
+        with tempfile.TemporaryDirectory() as tmpdir:
+            dst_repo = os.path.join(tmpdir, 'generalized_package')
+            shutil.copytree(src_repo, dst_repo)
+            rc = subprocess.run(['python', 'setup.py', 'sdist'],
+                check=True, cwd=dst_repo,
+                env={
+                    **os.environ,
+                    'TEST_PACKAGE_NAME': pkg_name,
+                    'TEST_PACKAGE_VERSION': '1.2.3.dev10+g123abc45',
```

Comment:
Maybe a comment explaining how this version can happen, or a pointer to the documentation that describes the version format ?

Note also that this test is somehow incomplete, because '1.2.3.dev10+g123abc45' is only one of the acceptable formats, the most complicated. Do we have tests for simpler formats?

There is also something that confuses me. Where is the actual upload in this test? In the call to out.out() ?We don't check for errors because it raises? I would still make it explicit with assert.NoRaise (or whatever the assert is called)

Reply:
To address the comments, I have added:
- comment about the version with local identifier
```
# Format: <public version identifier>[+<local version label>]
# See https://peps.python.org/pep-0440 for more details
```
simpler formats are quasi-tested with other tests. 

- added assert; `out.out` actually must return json with the version that the resource created (Concourse resource type APIs), so I am checking that it uploaded expected version
`assert output['version']['version'] == version`